runner/execer/os/os_execer.go

```diff
@@ -198,27 +220,26 @@ func (p *osProcess) Abort() (result execer.ProcessStatus) {
 }
 
 func memUsage(pid int) (execer.Memory, error) {
-	// Query for all sets of (pid, ppid, rss) and create a graph. Given a pid, walk the graph accruing memory.
+	// Query for all sets of (pid, sid, rss). Given a pid, find its associated sid.
+	// From there, sum the memory of all processes with the same session id.
 	// Note: look at the revision history for previous approaches which seemed inaccurate at times? ex:
 	//   str := "export P=%d; echo $(ps -orss= -p$(echo -n $(pgrep -g $P | tr '\n' ',')$P) | tr '\n' '+') 0 | bc"
 	str := `
 PID=%d
-PSLIST=$(ps -e -o pid= -o ppid= -o rss= | tr '\n' ';' | sed 's,;$,,')
+PSLIST=$(ps -e -o pid= -o sess= -o rss= | tr '\n' ';' | sed 's,;$,,')
```

Comment:
We may want to turn execer.Memory into a struct that allows us to flag how to track memory, i.e. by pgid on OSX and sid in aurora/linux. That or just use an env variable.

Similarly, when running on linux, we can prefer sid over pgid when killing processes.

Currently, it looks like this iterates over all system+user processes on OSX? Hmm how does this pass tests if it excessively overcounts memory - did I only check for growing memory in the tests and not cap the expected memory usage? Can you check?